pghrep/plugins/a003.go

```diff
@@ -1,15 +1,47 @@
 package main
 
-import "fmt"
+import (
+    "fmt"
+    "../src/pyraconv"
+    "../src/fmtutils"
+    "strconv"
+)
 
 var Data map[string]interface{}
 
 type prepare string
 
+func getSharefBuffersValue(hostSettings []interface{}) int64 {
+    for _, setting := range hostSettings {
+        curSetting := setting.(map[string]interface{})
+        if curSetting["name"] == "shared_buffers" {
+            if curSetting["unit"] != nil {
+                val, err := strconv.ParseInt(pyraconv.ToString(curSetting["setting"]), 10, 64)
+                if err != nil {
+                    return -1
```

Comment:
does `-1` mean something particular?

If we return `-1`, what will happen next? An error?

Reply:
BTW, why not `return nil` ?

Reply:
`plugins/a003.go:21:21: cannot use nil as type int64 in return argument`